### PR TITLE
Add allow_xpu to enable XPU UTs

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #130312

# Motivation
enable UTs under folder test/xpu/

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
Regarding the CI failure, it is due to torch-xpu-ops using a utility function `checkIndexTensorTypes`. But the utility function just changed its signature. We will rebase torch-xpu-ops to use the latest signature.

**Review by gujinghui — APPROVED** [review]

File: aten/src/ATen/native/mkldnn/xpu/Blas.cpp [hunk 1]
```diff
@@ -29,6 +29,10 @@ Tensor& addmm_out(
       "x",
       mat2.sizes()[1],
       ")");
+  TORCH_CHECK(
+    mat1.dtype() == mat2.dtype(),
```
Comment (guangyey) [inline aten/src/ATen/native/mkldnn/xpu/Blas.cpp:33]: align to other backend.

File: aten/src/ATen/native/mkldnn/xpu/Blas.cpp [hunk 2]
```diff
@@ -131,6 +135,10 @@ Tensor& mm_out(const Tensor& self, const Tensor& mat2, Tensor& result) {
       "x",
       mat2.sizes()[1],
       ")");
+  TORCH_CHECK(
+    self.dtype() == mat2.dtype(),
```
Comment (guangyey) [inline aten/src/ATen/native/mkldnn/xpu/Blas.cpp:139]: fix UT failure on test/xpu/test_gemm.py::test_mm_empty_inputs_mixed_dtype_errors

**Review by EikanWang — APPROVED** [review]

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**